pkg/provisioner/ironic/ironic.go

```diff
@@ -1295,3 +1326,83 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// Retrieve configuration in form of Ironic's clean steps
+func (p *ironicProvisioner) buildCleanSteps(ironicNode *nodes.Node) (cleanSteps []nodes.CleanStep) {
```

Comment:
The use of the term "clean" is a little confusing here. I see code on lines 1148-1152 to create clean steps for deprovisioning, and it looks like this method is only used at the beginning of the provisioning process. Is that correct? If so, we should name it in a way that indicates that.  Maybe something like buildProvisionCleanSteps()?